Tasks/Gradle/CodeAnalysis/SonarQube/common.ts

```diff
@@ -172,52 +71,108 @@ function isPrBuild(): boolean {
         sourceBranch.toLowerCase().startsWith("refs/pull/");
 }
 
+/**
+ * Returns, as an object, the contents of the report-task.txt file created by SonarQube plugins
+ * The returned object contains the following properties:
+ *     projectKey, serverUrl, dashboardUrl, ceTaskId, ceTaskUrl
+ * @param sonarPluginFolder The folder created by SonarQube integration during build
+ * @returns {SonarQubeRunSettings} An object with fields corresponding to the properties exposed in report-task.txt, or null if PR build
+ */
+function getSonarQubeRunSettings(sonarPluginFolder: string): SonarQubeRunSettings {
+    if (isPrBuild()) {
+        return null;
+    }
+
+    var reportFilePath:string = path.join(sonarPluginFolder, 'report-task.txt');
+    return SonarQubeRunSettings.createRunSettingsFromFile(reportFilePath);
+}
+
+/**
+ * Returns the appropriate SQMetrics object for this build.
+ * @param sqRunSettings An object with fields corresponding to the properties exposed in report-task.txt
+ * @returns {SonarQubeMetrics} An SQMetrics instance for this build on the SQ server, or null if PR build
+ */
+function getSonarQubeMetrics(sqRunSettings:SonarQubeRunSettings): SonarQubeMetrics {
+    if (isPrBuild()) {
+        return null;
+    }
+
+    var sqServer:ISonarQubeServer = new SonarQubeServer(SonarQubeEndpoint.createSonarQubeEndpoint());
+    return new SonarQubeMetrics(sqServer, sqRunSettings.ceTaskId);
+}
+
 function isNullOrEmpty(str) {
     return str === undefined || str === null || str.length === 0;
 }
 
 /**
- * Creates the string that comprises the build summary text, given the location of the /sonar build folder.
- * @param sqBuildFolder
- * @returns {any}
+ * Handle checking the status of the quality gate and failing the build if the quality gate has failed.
+ * No action is taken if the feature is not enabled, or sqRunSettings argument is null.
+ * @param sqRunSettings SonarQubeRunSettings object for this build or null if this is a PR build and the object could not be created
+ * @param sqMetrics     SonarQube metrics for the applicable run
+ * @returns {Q.Promise<void>} Promise resolved when action completes.
  */
-function createSonarQubeBuildSummary(sqBuildFolder:string):Q.Promise<string> {
-    // Task report is not created for PR builds - inform the user in the build summary
-    if (isPrBuild()) {
-        tl.debug('Pull request build, will not attempt to fetch detailed SonarQube report');
-        // Looks like: Detailed SonarQube reports are not available for pull request builds.
-        return Q.when(tl.loc('sqAnalysis_BuildSummaryNotAvailableInPrBuild'));
+function processSonarQubeBuildBreaker(sqRunSettings:SonarQubeRunSettings, sqMetrics:SonarQubeMetrics):Q.Promise<void> {
```

Comment:
Accepted, in progress. Will move all of the build breaker, build summary and isPR stuff into a class related to the VSTS server side.